x/trigger/module/module.go

```diff
@@ -0,0 +1,198 @@
+package trigger
+
+import (
+	"context"
+	"encoding/json"
+	"math/rand"
+
+	"github.com/gorilla/mux"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/spf13/cobra"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	cerrs "cosmossdk.io/errors"
+
+	sdkclient "github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+
+	triggerModule "github.com/provenance-io/provenance/x/trigger"
+	"github.com/provenance-io/provenance/x/trigger/client/cli"
+	"github.com/provenance-io/provenance/x/trigger/keeper"
+	"github.com/provenance-io/provenance/x/trigger/simulation"
+	"github.com/provenance-io/provenance/x/trigger/types"
+)
+
+var (
+	_ module.AppModule      = AppModule{}
+	_ module.AppModuleBasic = AppModuleBasic{}
+)
+
+// AppModuleBasic defines the basic application module used by the trigger module.
+type AppModuleBasic struct {
+	cdc codec.Codec
+}
+
+// Name returns the trigger module's name.
+func (AppModuleBasic) Name() string {
+	return types.ModuleName
+}
+
+// RegisterServices registers a gRPC query service to respond to the
+// module-specific gRPC queries.
+func (am AppModule) RegisterServices(cfg module.Configurator) {
+	types.RegisterMsgServer(cfg.MsgServer(), keeper.NewMsgServerImpl(am.keeper))
+	types.RegisterQueryServer(cfg.QueryServer(), am.keeper)
+}
```

Comment:
Nit: This is an `AppModule` method, but it's up with the `AppModuleBasic` methods. It'd be better placed lower down, somewhere after `NewAppModule`.